http-client/Network/HTTP/Client/Request.hs

```diff
@@ -111,16 +113,30 @@ getUri req = URI
     , uriFragment = ""
     }
 
+applyAnyUriBasedAuth :: URI -> Request -> Request
+applyAnyUriBasedAuth uri =  applyBasicAuth (username authInfo) (password authInfo)
+ where
+   authInfo = uriUserInfo $ fromJust $ uriAuthority uri
```

Comment:
That may be true today, but network-uri could change in the future without alerting us. I think your idea of `fromMaybe ""` could make sense here.

But I think I've got another problem with this: this code should only call applyBasicAuth if it actually finds a username and password, otherwise it's changing the behavior of existing programs incorrectly.